src/Test.FunctionalTests.BinSkim.Rules/RuleTests.cs

```diff
@@ -277,13 +278,23 @@ private BinaryAnalyzerContext CreateContext(TestMessageLogger logger, Properties
                 context.Rule = skimmer;
 
                 AnalysisApplicability applicability = skimmer.CanAnalyze(context, out string reasonForNotAnalyzing);
+
                 if (applicability != expectedApplicability)
                 {
                     sb.AppendLine("CanAnalyze did not correct indicate target applicability (unexpected return was " +
                         applicability + "): " +
                         Path.GetFileName(target));
                     continue;
                 }
+
+                if (expectedReasonForNotAnalyzing != null && reasonForNotAnalyzing != expectedReasonForNotAnalyzing)
```

Comment:
>expectedReasonForNotAnalyzing != null && reasonForNotAnalyzing != expectedReasonForNotAnalyzing [](http://example.com/codeflow?start=20&length=95)

why do we require the null check here??

Reply:
Not all tests that use this method take advantage of the additional functionality yet. We definitely want them to; but that should be part of a separate PR. We may encounter scenarios where the originally intended reasons for the test passing are no longer true; for that reason I did not pursue updating all tests yet.